GUI/AllegroInput.cpp

```diff
@@ -239,13 +239,63 @@ void AllegroInput::Update(void)
 	}
 	else 
 	{
-		m_MouseButtonsEvents[0] = mouse_b & AMBLEFT ? (m_MouseButtonsStates[0] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[0] == Down ? Released : None);
-		m_MouseButtonsEvents[1] = mouse_b & AMBMIDDLE ? (m_MouseButtonsStates[1] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[1] == Down ? Released : None);
-		m_MouseButtonsEvents[2] = mouse_b & AMBRIGHT ? (m_MouseButtonsStates[2] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[2] == Down ? Released : None);
-
-		m_MouseButtonsStates[0] = mouse_b & AMBLEFT ? Down : Up;
-		m_MouseButtonsStates[1] = mouse_b & AMBMIDDLE ? Down : Up;
-		m_MouseButtonsStates[2] = mouse_b & AMBRIGHT ? Down : Up;
+		// Left mouse button
+		if (mouse_b & AMBLEFT) {
+			if (m_MouseButtonsStates[0] == Up) { // Previously was up
+				m_MouseButtonsEvents[0] = Pushed;
+			}
+			else { // m_MouseButtonsStates[0] == Down
+				m_MouseButtonsEvents[0] = Repeat;
+			}
+			m_MouseButtonsStates[0] = Down;
+		}
+		else { // Left mouse button up
```

Comment:
I realize this file is still very much using old formatting, and I also realize I still haven't done what I said I would and put together a style guide, but our styling has else on the same line as the closing bracket as the if, i.e.
```
if (condition) {
	action;
} else {
	otherAction;
}
```